lib/request_handler/header_parser.rb

```diff
@@ -1,21 +1,49 @@
 # frozen_string_literal: true
 
+require 'request_handler/schema_parser'
 require 'request_handler/error'
+
 module RequestHandler
-  class HeaderParser
-    def initialize(env:)
+  class HeaderParser < SchemaParser
+    def initialize(env:, schema: nil, schema_options: {})
+      super(schema: schema, schema_options: schema_options) unless schema.nil?
+
       raise MissingArgumentError, env: 'is missing' if env.nil?
       @headers = Helper.deep_transform_keys_in_object(env.select { |k, _v| k.start_with?('HTTP_') }) do |k|
         k[5..-1].downcase.to_sym
       end
     end
 
     def run
-      headers
+      return headers if schema.nil?
+
+      validate_headers!
     end
 
     private
 
+    def validate_headers!
+      validate_schema(headers)
+    rescue SchemaValidationError => e
+      raise ExternalArgumentError, external_argument_error_params(e)
+    end
+
+    def external_argument_error_params(error)
+      error.errors.map do |schema_error|
+        header = schema_error[:source][:pointer]
+        {
+          status: '400',
+          code: "#{headers[header.to_sym] ? 'INVALID' : 'MISSING'}_HEADER",
+          detail: schema_error[:detail],
+          source: { header: format_header_name(header) }
```

Comment:
By jsonapi spec, `:source` can only contain param or pointer, so please remove it altogether.
Instead I'd prepend the header name to `:detail`, or even add a `:title` (maybe additionally).

Reply:
Good catch!